recipes/dbus/1.x.x/conanfile.py

```diff
@@ -0,0 +1,115 @@
+        "with_glib": [True, False],
+        "enable_assert": [True, False],
+        "enable_checks": [True, False],
+        "install_system_libs": [True, False]}
+
+    default_options = {
+        "with_x11": False,
+        "with_glib": False,
+        "enable_assert": False,
+        "enable_checks": False,
+        "install_system_libs": False}
+
+    generators = "cmake", "cmake_find_package"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    _cmake = None
+
+    def configure(self):
+        if self.settings.os not in ("Linux", "FreeBSD"):
+            raise ConanInvalidConfiguration("dbus is only supported on Linux")
+
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def requirements(self):
+        self.requires("expat/2.2.10")
+        if self.options.with_glib:
+            self.requires("glib/2.67.0")
+
+        if self.options.with_x11:
+            self.requires("xorg/system")
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+
+            self._cmake.definitions["DBUS_BUILD_TESTS"] = False
+            self._cmake.definitions["DBUS_ENABLE_DOXYGEN_DOCS"] = False
+            self._cmake.definitions["DBUS_ENABLE_XML_DOCS"] = False
+
+            self._cmake.definitions["DBUS_BUILD_X11"] = self.options.with_x11
+            self._cmake.definitions["DBUS_WITH_GLIB"] = self.options.with_glib
+            self._cmake.definitions["DBUS_DISABLE_ASSERT"] = not self.options.enable_assert
+            self._cmake.definitions["DBUS_DISABLE_CHECKS"] = not self.options.enable_checks
+            self._cmake.definitions["DBUS_INSTALL_SYSTEM_LIBS"] = self.options.install_system_libs
+
+            path_to_cmake_lists = os.path.join(self._source_subfolder, "cmake")
+
+            self._cmake.configure(source_folder=path_to_cmake_lists,
+                                  build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        dbus_cmake = tools.os.path.join(
+            self._source_subfolder, "cmake", "CMakeLists.txt")
+
+        if self.options.with_glib:
+            tools.replace_in_file(dbus_cmake, "GLib2", "glib")
+            tools.replace_in_file(dbus_cmake, "GLIB2", "GLIB")
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        for i in ["var", "share", "etc"]:
+            shutil.move(os.path.join(self.package_folder, i), os.path.join(self.package_folder, "res", i))
```

Comment:
Can setting `CMAKE_INSTALL_DATAROOTDIR` to `res` in `_configure_cmake`, avoid this move?

Reply:
no: https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12.20/cmake/bus/CMakeLists.txt#L103-104

Reply:
it's used only in https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12.20/cmake/modules/CPackInstallConfig.cmake so I'm removing the option